physics/CONV/progsigma_calc.f90

```diff
@@ -158,39 +170,33 @@ subroutine progsigma_calc (im,km,flag_init,flag_restart,flag_shallow,&
        do k = 2,km1
           do i = 1,im
              if(cnvflg(i))then
+               if(k >= kbcon1(i) .and. k < ktcon(i)) then
                 form(i,k)=-1.0*inbu(i,k)*(omega_u(i,k)*delt)
                 fdqb=0.5*((form(i,k)*zdqca(i,k)))
                 termC(i)=termC(i)+inbu(i,k)*   &
                      (fdqb+fdqa(i))*hvap*zeta(i,k)
                 fdqa(i)=fdqb
+               endif
              endif
          enddo
       enddo
 
       !sigmab
-      if(flag_init .and. .not. flag_restart)then
-         do i = 1,im
-            if(cnvflg(i))then
-               sigmab(i)=0.03
+      do i = 1,im
+         if(cnvflg(i))then
+            DEN=MIN(termC(i)+termB(i),1.E8)
```

Comment:
Same comment as above for all the fortran intrinsics in this block. (Explicitly setting the precision for these calls removes any ambiguity and makes to code more portable)

Reply:
@dustinswales Thanks! I've updated the code to replace the hardcoded precision constants with the appropriate precision.